### PR TITLE
Added wiggle sort in C-Plus-Plus

### LINKED ISSUE #2354: Wiggle Sort
## 🚀 Feature

Given an unsorted array arr , sort the array such that arr[0]<arr[1]>arr[2]<arr[3].. in C++ language

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

As part GSSOC'21

## Pitch

This algorithm is different from normal sorting. I can implement this in C++

### PR DESCRIPTION
**Pull request by aish2002**
Fixes #2354

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Wiggle sort is an algorithm to reorder an unsorted array in a particular way.
If given array is arr, Wriggle sort reorders the array such that arr[0]<arr[1]>arr[2]<arr[3]...
There can be more than one possible answer for a given input array

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue #2354

### CODE REVIEW

File: C-Plus-Plus/sort/wiggle_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/* WIGGLE SORT
+   This program accepts an array of unsorted numbers and
+   sorts the array such that arr[0]<arr[1]>arr[2]<arr[3]...
+
+*/
+
+#include <iostream>
+#include <vector>
+#include <algorithm>
```
Comment (siddharth25pandey) [inline C-Plus-Plus/sort/wiggle_sort.cpp]: Write single header file as **bits/stdc++**

File: C-Plus-Plus/sort/wiggle_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            cout << ", ";
+    }
+    cout << "]"
+         << "\n";
+}
+
+int main()
+{
+    int a, i, n;
+    cin >> n;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/sort/wiggle_sort.cpp]: Use proper naming convention

File: C-Plus-Plus/sort/wiggle_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    vector<int> arr;
+    for (int i = 0; i < n; i++)
+    { //The given array is stored in a vector arr
+        cin >> a;
```
Comment (siddharth25pandey) [inline C-Plus-Plus/sort/wiggle_sort.cpp]: Use proper naming convention

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@aish2002 Do the modifications as suggested.
Thanks!

**Comment by aish2002** [comment]
I have made necessary changes mentioned .@siddharth25pandey  kindly re-review the code

**Review by siddharth25pandey — APPROVED** [review]
@aish2002 LGTM, able to merge

**Comment by siddharth25pandey** [comment]
@all-contributors please add @aish2002 for his/her code

**Pull request merged**